### PR TITLE
PR: Add notebook tabs to the global file switcher

### LINKED ISSUE #69: Make Ctrl+P to show the file switcher here
I still have to think if/how we could do this. The problem is `Ctrl+P` is a global shortcut, not a plugin one.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #69 

Dependent of [this PR](https://github.com/spyder-ide/spyder/pull/4626)

### CODE REVIEW

File: spyder_notebook/notebookplugin.py [hunk 1]
```diff
@@ -386,9 +393,10 @@ def open_fileswitcher_dlg(self):
             self.fileswitcher_dlg.hide()
             self.fileswitcher_dlg.is_visible = False
             return
-        self.fileswitcher_dlg = FileSwitcher(self, self.tabwidget,
-                                             self.clients)
+        self.fileswitcher_dlg = FileSwitcher(self, self, self.tabwidget,
```
Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: Please remove this instance of the file switcher and its associated button. Given the changes in your Spyder PR, I think we don't need this anymore.

**Comment by ccordoba12** [comment]
Given that we haven't released this plugin yet, let's make it depend on Spyder 3.2 (in our `setup.py`).

I've also made progress in creating Spyder kernels for our notebook implementation, and that also requires changes to Spyder for that to work.

**Comment by ccordoba12** [comment]
@dalthviz, I changed CircleCI to use the 3.x Spyder branch. Now a test is failing, so please fix that so I can merge this PR.

**Comment by dalthviz** [comment]
@ccordoba12  the test that is failing is the one for the file switcher, since now the notebook doesn't have a instance of the file switcher, its relevant to have this test or should I removed it?

**Comment by ccordoba12** [comment]
Ok, then please remove that test.

**Comment by ccordoba12** [comment]
@dalthviz, let's use this icon

https://github.com/jupyterlab/jupyterlab/blob/master/packages/theming/style/icons/notebook.svg

for the file switcher entries of the notebook.

I think that's the last thing to do before merging this one ;-)

File: spyder_notebook/img_src/icon.svg [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ccordoba12) [inline spyder_notebook/img_src/icon.svg]: Let's put this icon in `images` instead of `img_src`.

File: spyder_notebook/notebookplugin.py [hunk 3]
```diff
@@ -37,7 +37,9 @@
 NOTEBOOK_TMPDIR = osp.join(TEMPDIR, 'notebooks')
 FILTER_TITLE = _("Jupyter notebooks")
 FILES_FILTER = "{} (*.ipynb)".format(FILTER_TITLE)
+PACKAGE_PATH = os.path.dirname(__file__)
```
Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: Please use `osp` instead of `os.path` here.

File: spyder_notebook/notebookplugin.py [hunk 4]
```diff
@@ -174,6 +170,9 @@ def register_plugin(self):
         self.focus_changed.connect(self.main.plugin_focus_changed)
         self.main.add_dockwidget(self)
         self.create_new_client(give_focus=False)
+        img_path = os.path.join(PACKAGE_PATH, 'img_src'+ os.sep +'icon.svg')
```
Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: img_path -> icon_path

This can rewritten as

    osp.join(PACKAGE_PATH, 'images', 'icon.svg')

Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: `osp.join` takes care of joining parts of a path with `osp.sep`.

**Pull request merged**